engine/src/main/java/org/hibernate/search/util/impl/ReflectionHelper.java

```diff
@@ -115,13 +115,34 @@ public static Object getMemberValue(Object bean, XMember getter) {
 	}
 
 	/**
-	 * Checks whether the specified class contains any Search specific annotations.
+	 * Checks whether the specified class contains any Search-specific annotations, i.e. if
+	 * the class itself or one of its superclasses carries such an annotation.
 	 *
 	 * @param mappedClass the {@code XClass} to check for Search annotations
 	 *
 	 * @return Returns {@code true} if the class contains at least one Search annotation, {@code false} otherwise
 	 */
 	public static boolean containsSearchAnnotations(XClass mappedClass) {
+		List<XClass> hierarchy = ReflectionHelper.createXClassHierarchy( mappedClass );
```

Comment:
Any reason why you added `ReflectionHelper.`? Might as well remove it so that we see it's a method of this very class?